tests/utils/object_store/integration_test_gs_object_store.py

```diff
@@ -0,0 +1,98 @@
+# Copyright 2022 MosaicML Composer authors
+# SPDX-License-Identifier: Apache-2.0
+
+import time
+from pathlib import Path
+
+import pytest
+
+from composer.utils import GCSObjectStore
+
+__DUMMY_OBJ__ = '/tmp/dummy.ckpt'
+__NUM_BYTES__ = 1000
+bucket_name = 'mosaicml-composer-tests'
+
+
+@pytest.fixture
+def gs_object_store():
+    remote_dir = 'gs://mosaicml-composer-tests/streaming/'
+    yield GCSObjectStore(remote_dir)
+
+
+# Todo: call trainer to generate a ckpt and resume the train from cloud
+def test_train_resuming():
+    pass
+
+
+def test_bucket_not_found():
+    with pytest.raises(FileNotFoundError):
+        _ = GCSObjectStore('gs://not_a_bucket/streaming')
+
+
+def test_get_uri(gs_object_store):
+    object_name = 'test-object'
+    expected_uri = 'gs://mosaicml-composer-tests/streaming/test-object'
+    assert (gs_object_store.get_uri(object_name) == expected_uri)
+
+
+def test_get_key(gs_object_store):
+    object_name = 'test-object'
+    expected_key = 'streaming/test-object'
+    assert (gs_object_store.get_key(object_name) == expected_key)
+
+
+@pytest.mark.parametrize('result', ['success', 'not found'])
```

Comment:
are the tests here and below supposed to be marked as remote only?

Reply:
Sure! How do I mark a test as remote? can you point me to an example? Thanks! @dakinggg

Reply:
I think these tests are not actually running because the filename does not start with `test`. So you should change the filename to start with `test` and then add `@pytest.mark.remote` as a decorator.

Reply:
@dakinggg  No I started the file name without "test" on purpose. Since we don't have a service account set up on the CI cluster yet, those tests will fail running on remote. 

I raised the request to upload the credential file to our CI cluster in this [ticket](https://mosaicml.atlassian.net/browse/IT-1826) but haven't got any updates.

Reply:
I'd suggest following the normal test format, and marking these as `@pytest.mark.remote`, and then also including `@pytest.skip` with a comment explaining why they are skipped for now.

Reply:
Agreed~ That is a better solution.